batch_job/src/batch_job_amazon_batch.c

```diff
@@ -0,0 +1,567 @@
+	debug(D_BATCH,"Creating the Job Definition: %s",tmp);
+        struct jx* jx = run_command(tmp);
+	free(tmp);
+	
+	char* arn = (char*)jx_lookup_string(jx,"jobDefinitionArn");
+	if(arn == NULL){
+		fatal("Fatal error when trying to create the job definition!");
+	}
+	jx_delete(jx);
+	
+	//now that we have create a job-definition, we can submit the job.
+	tmp = string_format("%s /usr/bin/time -f \"Submitting Job %s: %%e\" -a -o 'BatchJobAmazonBatchTimings.txt' aws batch submit-job --job-name %s --job-queue %s --job-definition %s_def",env_var,job_name,job_name,queue,job_name);
+	debug(D_BATCH,"Submitting the job: %s",tmp);
+	jx = run_command(tmp);
+	free(tmp);
+	char* jaid = strdup((char*)jx_lookup_string(jx,"jobId"));
+	if(!jaid)
+		fatal("NO JOB ID FROM AMAZON GIVEN");
+	jx_delete(jx);
+	return jaid;
+}
+enum{
+	DESCRIBE_AWS_JOB_SUCCESS = 1, //job exists, succeeded
+	DESCRIBE_AWS_JOB_FAILED = 0, //job exists, failed
+	DESCRIBE_AWS_JOB_NON_FINAL = -1, //exists, but in non-final state
+	DESCRIBE_AWS_JOB_NON_EXIST = -2 //job doesn't exist, should treat as a failure
+};
+
+static int finished_aws_job_exit_code(char* aws_jobid, char* env_var){
+	char* cmd = string_format("aws batch describe-jobs --jobs %s",aws_jobid);
+	struct jx* jx = run_command(cmd);
+	free(cmd);
+	struct jx* jobs_array = jx_lookup(jx,"jobs");
+	if(!jobs_array){
+		debug(D_BATCH,"Problem with given aws_jobid: %s",aws_jobid);
+		return DESCRIBE_AWS_JOB_NON_EXIST;
+	}
+	struct jx* first_item = jx_array_index(jobs_array,0);
+	if(!first_item){
+		debug(D_BATCH,"Problem with given aws_jobid: %s",aws_jobid);
+		return DESCRIBE_AWS_JOB_NON_EXIST;
+	}
+	int ret = (int)jx_lookup_integer(first_item,"exitCode");
+	jx_delete(jx);
+	return ret;
+}
+
+static int describe_aws_job(char* aws_jobid, char* env_var){
+	char* cmd = string_format("aws batch describe-jobs --jobs %s",aws_jobid);
+	struct jx* jx = run_command(cmd);
+	free(cmd);
+	int succeed = DESCRIBE_AWS_JOB_NON_FINAL; //default status
+	struct jx* jobs_array = jx_lookup(jx,"jobs");
+	if(!jobs_array){
+		debug(D_BATCH,"Problem with given aws_jobid: %s",aws_jobid);
+		return DESCRIBE_AWS_JOB_NON_EXIST;
+	}
+	struct jx* first_item = jx_array_index(jobs_array,0);
+	if(!first_item){
+		debug(D_BATCH,"Problem with given aws_jobid: %s",aws_jobid);
+		return DESCRIBE_AWS_JOB_NON_EXIST;
+	}
+	if(strstr((char*)jx_lookup_string(first_item,"status"),"SUCCEEDED")){
+		succeed = DESCRIBE_AWS_JOB_SUCCESS;
+	}
+	if(strstr((char*)jx_lookup_string(first_item,"status"),"FAILED")){
+		succeed = DESCRIBE_AWS_JOB_FAILED;
+	}
+
+	//start and stop
+	if(succeed == DESCRIBE_AWS_JOB_SUCCESS || succeed == DESCRIBE_AWS_JOB_FAILED){	
+		int64_t created_string = (int64_t) jx_lookup_integer(first_item,"createdAt");
```

Comment:
I don't think you need these casts.

Reply:
the return is jx_int_t and I felt it safe to be explicit about the casting, so the compiler wouldn't complain.